File: analyzers/tests/SonarAnalyzer.Test/TestCases/Utilities/MethodDeclarationsAnalyzer/MethodDeclarations.cs
namespace Samples;

public class Address
{
    public string GetZipCode() => "12345";
}

public struct Person
{
    public string GetFullName() => "John Doe";
}

public record class Company
{
    public string GetCompanyName() => "ACME";
}

public record struct Employee
{
    public string GetEmployeeName() => "Jane Doe";
}

public enum EnumDeclaration
{
    Value1,
    Value2
}

public class Visibility
{
    public void PublicMethod() { }

    protected internal void ProtectedInternalMethod() { }

    protected void ProtectedMethod() { }

    internal void InternalMethod() { }

    private protected void PrivateProtectedMethod() { }

    private void PrivateMethod() { }

    void NoAccessModifierMethod() { }

    internal class InternalClass
    {
        public void Method() { }
    }

    private class PrivateClass
    {
        public void Method() { }
    }
}

file class FileClass
{
    public void Method() { }
}

class NoModifiers
{
    public void Method() { }
}

public class MultipleMethods
{
    public void Method1() { }

    public void Method2() { }
}

public class Overloads
{
    public void Method() { }

    public void Method(int i) { }

    public void Method(string s) { }

    public void Method(int i, string s) { }
}

public class GenericClass<T>
{
    public void Method() { }

    public void Method<U>() { }
}

public class WithGenericMethod
{
    public void Method<T>() { }
}

public partial class PartialClass
{
    public void InFirstFile() { }

    public partial void PartialMethod();
}

public class PropertiesAndIndexers
{
    private int[] values;

    public string Property { get; set; }

    public int this[int i]
    {
        get => values[i];
        set => values[i] = value;
    }
}

public class LocalFunctions
{
    public void Main()
    {
        void LocalFunction() { }
    }
}

File: analyzers/tests/SonarAnalyzer.Test/TestCases/Utilities/MethodDeclarationsAnalyzer/MethodDeclarations.vb
Namespace Samples

    Public Class Address
        Public Function GetZipCode() As String
            Return "12345"
        End Function
    End Class

    Public Structure Person
        Public Function GetFullName() As String
            Return "John Doe"
        End Function
    End Structure

    Public Enum EnumDeclaration
        Value1
        Value2
    End Enum

    Public Class Visibility
        Public Sub PublicMethod()
        End Sub

        Protected Friend Sub ProtectedFriendMethod()
        End Sub

        Protected Sub ProtectedMethod()
        End Sub

        Friend Sub FriendMethod()
        End Sub

        Private Protected Sub PrivateProtectedMethod()
        End Sub

        Private Sub PrivateMethod()
        End Sub

        Sub NoAccessModifierMethod()
        End Sub

        Friend Class FriendClass
            Public Sub Method()
            End Sub
        End Class

        Private Class PrivateClass
            Public Sub Method()
            End Sub
        End Class
    End Class

    Class NoModifiers
        Public Sub Method()
        End Sub
    End Class

    Public Class MultipleMethods
        Public Sub Method1()
        End Sub

        Public Sub Method2()
        End Sub
    End Class

    Public Class OverloadedMethods
        Public Sub Method()
        End Sub

        Public Sub Method(i As Integer)
        End Sub

        Public Sub Method(s As String)
        End Sub

        Public Sub Method(i As Integer, s As String)
        End Sub
    End Class

    Public Class GenericClass(Of T)
        Public Sub Method()
        End Sub

        Public Sub Method(Of U)()
        End Sub
    End Class

    Public Class WithGenericMethod
        Public Sub Method(Of T)()
        End Sub
    End Class

    Partial Public Class PartialClass
        Public Sub InFirstFile()
        End Sub

        Private Partial Sub PartialMethod()
        End Sub
    End Class

    Public Class PropertiesAndIndexers
        Private values() As Integer

        Public Property [Property] As String

        Default Public Property Item(index As Integer) As Integer
            Get
                Return values(index)
            End Get
            Set(value As Integer)
                values(index) = value
            End Set
        End Property
    End Class

End Namespace
